File: backend/src/Naheulbook.Web/Responses/CharacterLevelUpResponse.cs
using JetBrains.Annotations;
using Naheulbook.Shared.TransientModels;

namespace Naheulbook.Web.Responses;

[PublicAPI]
public class CharacterLevelUpResponse
{
    public List<ActiveStatsModifier> NewModifiers { get; set; } = null!;
    public List<Guid> NewSkillIds { get; set; } = null!;
    public int NewLevel { get; set; }
    public List<SpecialityResponse> NewSpecialities { get; set; } = null!;
}

File: frontend/src/app/api/responses/character-level-up-response.ts
import {Guid} from '../shared/util';
import {SpecialityResponse} from './speciality-response';

export interface CharacterLevelUpResponse {
    newModifiers: ActiveStatsModifierResponse[];
    newSkillIds: Guid[];
    newSpecialities: SpecialityResponse[];
    newLevel: number;
}

export type ActiveStatsModifierResponse = {
    // FIXME: when needed
}

